Mathlib/SetTheory/Ordinal/Arithmetic.lean

```diff
@@ -1123,27 +1124,52 @@ theorem bddAbove_range {ι : Type u} (f : ι → Ordinal.{max u v}) : BddAbove (
     exact le_of_lt (Cardinal.lt_ord.2 ((lt_succ _).trans_le
       (le_ciSup (Cardinal.bddAbove_range.{_, v} _) _)))⟩
 
+instance small_Iio (o : Ordinal.{u}) : Small.{u} (Set.Iio o) :=
+  let f : o.toType → Set.Iio o :=
+    fun x => ⟨typein (α := o.toType) (· < ·) x, typein_lt_self x⟩
+  let hf : Surjective f := fun b =>
+    ⟨enum (α := o.toType) (· < ·) ⟨b.val,
+        by
+          rw [type_lt]
+          exact b.prop⟩,
+      Subtype.ext (typein_enum _ _)⟩
+  small_of_surjective hf
+
+instance small_Iic (o : Ordinal.{u}) : Small.{u} (Set.Iic o) := by
+  rw [← Iio_succ]
+  infer_instance
+
+theorem bddAbove_of_small (s : Set Ordinal.{u}) [h : Small.{u} s] : BddAbove s := by
+  obtain ⟨a, ha⟩ := bddAbove_range (fun x => ((@equivShrink s h).symm x).val)
+  use a
+  intro b hb
+  simpa using ha (mem_range_self (equivShrink s ⟨b, hb⟩))
+
+theorem bddAbove_iff_small {s : Set Ordinal.{u}} : BddAbove s ↔ Small.{u} s :=
+  ⟨fun ⟨a, h⟩ => small_subset <| show s ⊆ Iic a from fun _ hx => h hx, fun _ =>
+    bddAbove_of_small _⟩
+
 /-- `le_ciSup` whenever the outputs live in a higher universe than the inputs. -/
-protected theorem le_iSup {ι : Type u} (f : ι → Ordinal.{max u v}) : ∀ i, f i ≤ iSup f :=
-  le_ciSup (bddAbove_range f)
+protected theorem le_iSup {ι} (f : ι → Ordinal.{u}) [Small.{u} ι] : ∀ i, f i ≤ iSup f :=
```

Comment:
It's quite unfortunate that this lemma is worse at inferring the function argument, though I think it's still an improvement to have it like this nevertheless. Can you perhaps explain this quirk in the docstring?